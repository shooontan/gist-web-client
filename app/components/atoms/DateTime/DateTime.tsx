import * as React from 'react';
import pure from 'recompose/pure';
import styled from 'styled-components';

interface Props extends React.TimeHTMLAttributes<HTMLTimeElement> {}

const DateTime = (props: Props) => {
  const { children, ...rest } = props;
  return <Time {...rest}>{children}</Time>;
};

export default pure(DateTime);

const Time = styled.time`
  color: #666;
  font-size: 12px;
`;
