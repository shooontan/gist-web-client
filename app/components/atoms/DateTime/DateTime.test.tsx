import { render } from 'react-testing-library';
import DateTime from './DateTime';

const datetime = '2018-10-15T12:09:50Z';

test('snapshot DateTime', () => {
  const { container } = render(<DateTime />);
  expect(container.firstChild).toMatchSnapshot();

  const { container: datetimeContainer } = render(
    <DateTime>{datetime}</DateTime>
  );
  expect(datetimeContainer.firstChild).toMatchSnapshot();

  const { container: attrComtainer } = render(
    <DateTime dateTime={datetime}>{datetime}</DateTime>
  );
  expect(attrComtainer.firstChild).toMatchSnapshot();
});
