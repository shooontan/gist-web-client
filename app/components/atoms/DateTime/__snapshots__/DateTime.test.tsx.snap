// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot DateTime 1`] = `
.c0 {
  color: #666;
  font-size: 12px;
}

<time
  class="c0"
/>
`;

exports[`snapshot DateTime 2`] = `
.c0 {
  color: #666;
  font-size: 12px;
}

<time
  class="c0"
>
  2018-10-15T12:09:50Z
</time>
`;

exports[`snapshot DateTime 3`] = `
.c0 {
  color: #666;
  font-size: 12px;
}

<time
  class="c0"
  datetime="2018-10-15T12:09:50Z"
>
  2018-10-15T12:09:50Z
</time>
`;
